File: apps/connector/src/types/health.ts
export type UiHealthStatus = "online" | "degraded" | "offline";

export interface HealthSummary {
  latencyMs: number;
  tunnelConnected: boolean;
  gatewayOk: boolean;
  consecutiveFailures: number;
}

export function mapHealthSummary(summary: HealthSummary): UiHealthStatus {
  if (summary.consecutiveFailures >= 3) {
    return "offline";
  }
  if (summary.tunnelConnected && summary.gatewayOk) {
    return "online";
  }
  return "degraded";
}

File: apps/connector/src-tauri/src/health.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum HealthStatus {
    Online,
    Degraded,
    Offline,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct HeartbeatSample {
    pub latency_ms: u64,
    pub tunnel_connected: bool,
    pub gateway_ok: bool,
}
